src/test/regress/sql/multi_extension.sql

```diff
@@ -9,18 +9,18 @@
 
 SET citus.next_shard_id TO 580000;
 
-CREATE SCHEMA test;
-
+SELECT $definition$
 CREATE OR REPLACE FUNCTION test.maintenance_worker(p_dbname text DEFAULT current_database())
     RETURNS pg_stat_activity
     LANGUAGE plpgsql
 AS $$
 DECLARE
    activity record;
 BEGIN
+    DO 'BEGIN END'; -- Force maintenance daemon to start
```

Comment:
[This function in multi_utilities](https://github.com/citusdata/citus/blob/master/src/test/regress/sql/multi_utilities.sql#L142-L174) is similar and has a timeout after which the call fails. Maybe it's good to add that here too.

Reply:
I don't think it's necessary here

Reply:
I don't think I provided enough context here. It is indeed not needed here since this loop loops indefinitely. However, in the new loops you added below it is needed, since they loop 100 times at most. If we ever change anything so that the early exit condition does not hold, we will only notice test times increasing (which is hard to notice).

I added the failure exit of the loop I mentioned in my previous comment in this recent PR: https://github.com/citusdata/citus/pull/3507. It decreased test times by 20 seconds, because before we did not notice that the exit condition was incorrect.